retrofit-mock/src/main/java/retrofit/MockRetrofit.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2013 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package retrofit;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+/**
+ * Creates {@link Call} instances which simulate the delay and error characteristics of a real
+ * network.
+ * <p>
+ * Because APIs are defined as interfaces, versions of the API that use mock data can be created by
+ * simply implementing the API interface on a class. These mock implementations execute
+ * synchronously which is a large deviation from the behavior of those backed by an API call over
+ * the network. By using the {@code Call} instances created using this class, the interface will
+ * still use mock data but exhibit the delays and errors of a real network.
+ * <p>
+ * Example:
+ * <pre>
+ * public interface UserService {
+ *   &#64;GET("/user/{id}")
+ *   Call&lt;User> getUser(@Path("id") String userId);
+ * }
+ *
+ * public class MockUserService implements UserService {
+ *   private final MockRetrofit mock = // ...
+ *
+ *   &#64;Override public Call&lt;User> getUser(String userId) {
+ *     return mock.newSuccessCall(new User("Jake"));
+ *   }
+ * }
+ * </pre>
+ */
+public final class MockRetrofit {
```

Comment:
Also: Snapshots for `retrofit-mock` and the samples are out-dated.

- https://oss.sonatype.org/content/repositories/snapshots/com/squareup/retrofit/retrofit-mock/2.0.0-SNAPSHOT/
- https://oss.sonatype.org/content/repositories/snapshots/com/squareup/retrofit/samples/